misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/coverage/CoverageDataFormatter.java

```diff
@@ -0,0 +1,347 @@
+/*
+ *  Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package org.ballerinalang.testerina.coverage;
+
+import org.ballerinalang.bre.coverage.ExecutedInstruction;
+import org.ballerinalang.config.ConfigRegistry;
+import org.ballerinalang.testerina.core.entity.Test;
+import org.ballerinalang.testerina.core.entity.TestSuite;
+import org.ballerinalang.testerina.util.Constants;
+import org.ballerinalang.util.codegen.LineNumberInfo;
+import org.ballerinalang.util.debugger.LineNumberInfoHolder;
+import org.ballerinalang.util.debugger.PackageLineNumberInfo;
+import org.ballerinalang.util.exceptions.BallerinaException;
+import org.wso2.ballerinalang.compiler.util.Names;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.PrintStream;
+import java.io.Writer;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This is for converting the Ip coverage information to output format.
+ * Formats: lcov
+ *
+ * @since 0.985
+ */
+public class CoverageDataFormatter {
+
+    private static PrintStream outStream = System.out;
+
+    private Map<String, LineNumberInfoHolder> lineNumberInfoHolderForProject;
+
+    private String testName = null;
+
+    private static final String COVERAGE_DATA_FILE_NAME = "coverage.data.file.name";
+
```

Comment:
Shall we remove additional empty lines here?

Reply:
These empty lines are removed in an logically grouping way (some are removed)